obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1071 @@
+    'g': 'ground truth',
+    'o': 'other',
+}
+
+
+# Convert GSE2 to ObsPy event types
+EVENT_TYPES = {
+    'ke': (EventTypeCertainty.KNOWN,
+           EventType('earthquake')),
+    'se': (EventTypeCertainty.SUSPECTED,
+           EventType('earthquake')),
+    'kr': (EventTypeCertainty.KNOWN,
+           EventType('rock burst')),
+    'sr': (EventTypeCertainty.SUSPECTED,
+           EventType('rock burst')),
+    'ki': (EventTypeCertainty.KNOWN,
+           EventType('induced or triggered event')),
+    'si': (EventTypeCertainty.SUSPECTED,
+           EventType('induced or triggered event')),
+    'km': (EventTypeCertainty.KNOWN,
+           EventType('mining explosion')),
+    'sm': (EventTypeCertainty.SUSPECTED,
+           EventType('mining explosion')),
+    'kx': (EventTypeCertainty.KNOWN,
+           EventType('experimental explosion')),
+    'sx': (EventTypeCertainty.SUSPECTED,
+           EventType('experimental explosion')),
+    'kn': (EventTypeCertainty.KNOWN,
+           EventType('nuclear explosion')),
+    'sn': (EventTypeCertainty.SUSPECTED,
+           EventType('nuclear explosion')),
+    'ls': (EventTypeCertainty.KNOWN,
+           EventType('landslide')),
+    'uk': (None,
+           EventType('other')),
+}
+
+
+# Convert GSE2 to ObsPy pick onset
+PICK_ONSETS = {
+    'E': PickOnset.EMERGENT,
+    'I': PickOnset.IMPULSIVE,
+    'Q': PickOnset.QUESTIONABLE,
+}
+
+
+def _is_gse2(filename):
+    """
+    Checks whether a file is GSE2.0 format.
+
+    :type filename: str
+    :param filename: Name of the GSE2.0 file to be checked.
+    :rtype: bool
+    :return: ``True`` if GSE2.0 file.
+    """
+    try:
+        with open(filename, 'rb') as fh:
+            temp = fh.read(12)
+    except:
+        return False
+    if temp != b'BEGIN GSE2.0':
+        return False
+    return True
+
+
+class LinesIterator(object):
+    """
+    Iterator to iterate file lines and count lines. Usefull for warning
+    messages.
+    """
+    def __init__(self, lines):
+        self.lines = iter(lines)
+        self.line_nb = 0
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self.line_nb += 1
+        return next(self.lines)
+
+
+class Unpickler(object):
+    """
+    De-serialize a GSE2.0 string into an ObsPy Catalog object.
+    """
+
+    def __init__(self, inventory, default_network_code, default_location_code,
+                 default_channel_code, res_id_prefix, fields,
+                 event_point_separator, agency):
+        self.default_network_code = default_network_code
+        self.default_location_code = default_location_code
+        self.default_channel_code = default_channel_code
+        self.inventory = inventory
+        self.res_id_prefix = res_id_prefix
+        self.event_point_separator = event_point_separator
+        self.agency = agency
+        self.author = ''
+
+        self.fields = {
+            'line_1': {
+                'time': slice(0, 21),
```

Comment:
Very clean with the `slice` objects :-) I'll be doing that in the future as well!